.github/workflows/actions.yml

```diff
@@ -0,0 +1,114 @@
+name: innovativeproject-loancalculator
+
+on:
+  push:
+    branches: '**'
+  pull_request:
+    branches: '**'
+
+jobs:
+  lint-backend:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.9.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.9.10"
+    
+    - name: Install backend dependencies
```

Comment:
Please extract this step and execute it once per pipeline 
Here's an example how you can use caching mechanism
https://stackoverflow.com/questions/62977821/how-to-cache-poetry-install-for-github-actions

Reply:
Done on github-actions-enhanced branch